### PR TITLE
Add a run_tests flag that captures core dumps from engine unit tests and runs a GDB script

### PR DESCRIPTION
**Pull request by jason-simmons**
The output of the GDB script will be placed in a directory that can be
uploaded to cloud storage by the LUCI recipe.

### CODE REVIEW

File: testing/run_tests.py [hunk 1]
```diff
@@ -11,6 +11,8 @@
 import glob
 import os
 import re
+import resource
```
Comment (zanderso) [inline testing/run_tests.py:14]: Looks like this doesn't exist on Windows.

Comment (jason-simmons) [reply testing/run_tests.py]: Fixed this to only set the core dump ulimit on Linux

(In theory this approach could also work on macOS.  But the directory that holds core dumps on macOS is not writable by the jobs on the LUCI bots).

File: testing/run_tests.py [hunk 2]
```diff
@@ -134,12 +135,33 @@ def RunEngineExecutable(build_dir, executable_name, filter, flags=[],
   else:
     test_command = [ executable ] + flags
 
-  RunCmd(test_command, cwd=cwd, forbidden_output=forbidden_output, expect_failure=expect_failure, env=env)
 
-
-def RunCCTests(build_dir, filter, coverage):
+  try:
```
Comment (zanderso) [inline testing/run_tests.py:139]: nit: remove extra newline above.

File: testing/run_tests.py [hunk 3]
```diff
@@ -134,12 +135,33 @@ def RunEngineExecutable(build_dir, executable_name, filter, flags=[],
...
+    if luci_test_outputs_path and os.path.exists(core_path) and os.path.exists(unstripped_exe):
+      dump_path = os.path.join(luci_test_outputs_path, '%s_%s.txt' % (executable_name, sys.platform))
+      print 'Writing core dump analysis to %s' % dump_path
+      subprocess.call([
+        os.path.join(buildroot_dir, 'flutter', 'testing', 'analyze_core_dump.sh'),
+        buildroot_dir, unstripped_exe, core_path, dump_path,
+      ])
+      os.unlink(core_path)
+    raise
+
```
Comment (zanderso) [inline testing/run_tests.py:157]: nit: need two newlines between functions

**Review by zanderso — APPROVED** [review]
lgtm w/ nits

**Pull request merged**